aspnetcore/grpc/troubleshoot.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Troubleshoot gRPC on .NET Core
+author: jamesnk
+description: Troubleshoot errors when using gRPC on .NET Core.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: jamesnk
+ms.custom: mvc
+ms.date: 08/12/2019
+uid: grpc/troubleshoot
+---
+# Troubleshoot gRPC on .NET Core
+
+By [James Newton-King](https://twitter.com/jamesnk)
+
+## Mismatch between client and service SSL/TLS configuration
+
+The gRPC template and samples use [Transport Layer Security (TLS)](https://tools.ietf.org/html/rfc5246) to secure gRPC services by default. gRPC clients need to use a secure connection to call secured gRPC services successfully.
+
+You can verify the ASP.NET Core gRPC service is using TLS in the logs written on app start. The service will be listening on `https`:
+
+```
+info: Microsoft.Hosting.Lifetime[0]
+      Now listening on: https://localhost:5001
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Development
+```
+
+The .NET Core client must use `https` in the server address to make calls with a secured connection:
+
+```csharp
+static async Task Main(string[] args)
+{
+    var httpClient = new HttpClient();
+    // The port number(5001) must match the port of the gRPC server.
+    httpClient.BaseAddress = new Uri("https://localhost:5001");
+    var client = GrpcClient.Create<Greeter.GreeterClient>(httpClient);
+}
+```
+
+All gRPC client implementations support TLS. gRPC clients from other languages typically require the channel configured with `SslCredentials`. `SslCredentials` specifies the certificate that the client will use, and it must be used instead of insecure credentials. See [gRPC Authentication](https://www.grpc.io/docs/guides/auth/) for samples of configuring the different gRPC client implementations to use TLS.
```

Comment:
```suggestion
All gRPC client implementations support TLS. gRPC clients from other languages typically require the channel configured with `SslCredentials`. `SslCredentials` specifies the certificate that the client will use, and it must be used instead of insecure credentials. For examples of configuring the different gRPC client implementations to use TLS, see [gRPC Authentication](https://www.grpc.io/docs/guides/auth/).
```